### PR TITLE
Make E2E env setup more robust; add GHA reporter for E2E and JS unit tests

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

- Adds a GitHub Action reporter to Jest unit and E2E configs
- Clean up AMP settings and trash all posts before running E2E tests

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: bin/local-env/docker-compose.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ services:
     command: tail -f /dev/null
 
   mysql:
-    image: mysql:5.7
+    image: mariadb:10
```
Comment (pierlon) [inline bin/local-env/docker-compose.yml:31]: The `mysql` Docker image doesn't have an arm64 compatible build yet, but `mariadb` does so I went with that.

File: package.json [hunk 2]
```diff
@@ -59,7 +59,7 @@
     "@wordpress/hooks": "2.12.3",
     "@wordpress/jest-puppeteer-axe": "3.0.3",
     "@wordpress/plugins": "3.0.0",
-    "@wordpress/scripts": "15.0.1",
+    "@wordpress/scripts": "16.0.0",
```
Comment (pierlon) [inline package.json:62]: Updated `@wordpress/scripts` to v16 as it uses `puppeteer-core` v9, which has support for Apple M1.

**Comment by pierlon** [comment]
The "Unit test: PHP 8.0, WP trunk" PR check is currently failing with:

> PHP Fatal error:  Cannot redeclare construct_wp_query_args() (previously declared in /tmp/wordpress/src/wp-includes/blocks.php:978) in /tmp/wordpress/src/wp-content/plugins/gutenberg/lib/query-utils.php on line 20

I see that https://github.com/WordPress/gutenberg/pull/32008 was merged recently to resolve the issue, and should be apart of Gutenberg v10.7 set to be released today (according to the [milestone](https://github.com/WordPress/gutenberg/milestone/149) due date).

**Comment by pierlon** [comment]
> I see that WordPress/gutenberg#32008 was merged recently to resolve the issue...

Bug fix was shipped as apart of the 10.6.2 release: https://github.com/WordPress/gutenberg/releases/tag/v10.6.2.

**Review by delawski — APPROVED** [review]
The changes look well to me. I've tried out the E2E tests locally and they work as expected.

**Comment by westonruter** [comment]
@pierlon Is the Percy failure due to there not being a previous snapshot to compare against?

**Comment by pierlon** [comment]
@westonruter It seems to be due to the work being done in #6282 since that PR is using this branch as the base. It can be ignored for now.

**Comment by westonruter** [comment]
Cool. The conflicts in `package-lock.json` just need to be resolved.

**Comment by pierlon** [comment]
The "Build, test & measure / Unit test: PHP 8.0, WP trunk" PR check is failing once again, but with a different error:


> PHP Fatal error:  Cannot declare class WP_Theme_JSON, because the name is already in use in /tmp/wordpress/src/wp-content/plugins/gutenberg/lib/class-wp-theme-json.php on line 12

We can ignore that error for now as it has been resolved via https://github.com/WordPress/gutenberg/pull/32059 and should be available in the upcoming Gutenberg 10.7 release.

**Comment by westonruter** [comment]
So we should wait for Gutenberg 10.7 to be released?

**Comment by pierlon** [comment]
Nope, shouldn't be considered a blocker since this issue only occurs on WP trunk.

**Comment by westonruter** [comment]
But we currently have `trunk` as a required check.

**Comment by pierlon** [comment]
I don't think that should be the case since it's expected that tests will fail from time to time, especially in this case where the failure is due to external factors not related to the plugin.

**Comment by westonruter** [comment]
OK, I removed trunk from the checks required to pass.

**Comment by westonruter** [comment]
Moving this to 2.2 as I don't think we need this on the 2.1 branch.

**Pull request merged**